packages/web-components/src/component-mixins/parent-visibility/parent-visibility.ts

```diff
@@ -45,15 +44,25 @@ const ParentVisibilityMixin = <T extends Constructor<HTMLElement>>(Base: T) => {
     }
 
     /**
-     * A list of potential parent components that may be hide their content on
+     * A list of potential parent components that may hide their content on
      * first render. Lists event names that indicate the parent element
      * visibility has changed keyed by component selector strings.
      */
     static get parentsThatHide() {
+      const { eventTabSelected } = this
+        .constructor as typeof ParentVisibilityMixinImpl;
       return {
-        [`${c4dPrefix}-tab`]: C4DTab.eventTabSelected,
+        [`${c4dPrefix}-tab`]: eventTabSelected,
       };
     }
+
+    /**
+     * The name of the custom event that fires whenever a tab's selected state
+     * changes.
+     */
+    static get eventTabSelected() {
+      return `${c4dPrefix}-tab-selected`;
+    }
```

Comment:
Do you think it would be too over-engineered to define this string in `/src/components/tabs-extended/defs.ts` then import that value into both this file and the tab source file? I'm worried about the event name changing in the future and there being no indication that it needs to change here, too, causing things to silently break.

Reply:
I like that suggestion, lets do that.

I was going off existing precedant in places like the `video-player` component where event names are repeated like this (see below). But I agree with you that it's better to have it be defined in one place.

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/video-player/video-player.ts#L288-L290
https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/main/packages/web-components/src/components/video-player/video-player-composite.ts#L320-L322